src/main.js

```diff
@@ -1806,40 +1813,20 @@ const DiscoveryFeedApplication = new Lang.Class({
         if (this._debugWindow)
             return;
 
-        // There seems to be a race condition with the WM that can
-        // lead the sidebar into an inconsistent state if the
-        // _onActiveWindowChanged callback gets executed in such a
-        // way that ends up calling to hide() between the user pressed
-        // the tray button and the sidebar has been made visible,
-        // which can lead to the sidebar never been displayed.
-        this._window.connect('map-event', Lang.bind(this, function() {
-            if (this._changedSignalId == -1) {
-                this._changedSignalId = Wnck.Screen.get_default().connect('active-window-changed',
-                                                                          Lang.bind(this, this._onActiveWindowChanged));
-            }
-            return false;
-        }));
-    },
-
-    _onActiveWindowChanged: function() {
-        let active_window = Wnck.Screen.get_default().get_active_window();
-        let current_window = this._window.get_window();
-        let active_window_xid = active_window ? active_window.get_xid() : 0;
-        let current_window_xid = current_window ? current_window.get_xid() : 0;
-
-        if (active_window !== null) {
-            // try to match transient windows
-            let transient_window = active_window.get_transient();
-
-            if (transient_window !== null &&
-                current_window_xid === transient_window.get_xid()) {
-                return;
-            }
-        }
+        // Connecting to 'draw' here is not ideal, however it seems like
+        // the 'realize' signal is not fired for GtkWindow when it gets
+        // mapped
```

Comment:
I'd be very surprised if that was the case... Seems more likely that `realize` is somehow already called before you get here. Perhaps as a consequence of this call? https://github.com/endlessm/eos-discovery-feed/blob/9ce002d8d0a8dace4cf0960ca20a9837756df47e/src/main.js#L1698

You could also probably override `vfunc_realize()`...